internal/dag/cache.go

```diff
@@ -94,8 +95,10 @@ func (kc *KubernetesCache) matchesIngressClass(obj *networking_v1.IngressClass)
 	}
 	// Otherwise, the name of the ingress class must match what has been
 	// configured.
-	if obj.Name == kc.IngressClassName {
-		return true
+	for _, contourIngressClassEntry := range strings.Split(kc.IngressClassName, ",") {
```

Comment:
I don't think we're actually using the DAG cache's `ingressclass` field (https://github.com/projectcontour/contour/blob/81cbcabd93b5051e6c46ae0f122089a4fb8af926/internal/dag/cache.go#L54), which this function `matchesIngressClass` provides a filter for. Since Contour can now process >1 ingress class, I'd rather drop the field entirely, and this function, if it's truly not being used; otherwise, it would make sense that it'd need to be converted to a map or slice to support multiple ingress classes.

Reply:
I don't feel confident making DAG changes beyond simple ones like this.  Could this cleanup be deferred to a different MR?

Reply:
Yeah, we can do that, I'll create an issue for this.